src/lib/transition_frontier_controller_tests/test_bootstrap_controller.ml

```diff
@@ -37,6 +37,86 @@ end)
 
 let%test_module "Bootstrap Controller" =
   ( module struct
+    let%test "`bootstrap_controller` caches all transitions it is passed \
+              through the `transition_reader` pipe" =
+      let transition_graph =
+        Bootstrap_controller.For_tests.Transition_cache.create ()
+      in
+      let max_length = 4 in
+      let num_breadcrumbs = 10 in
+      let logger = Logger.create () in
+      let network = Network.create ~logger in
+      let ancestor_prover = Ancestor.Prover.create ~max_size:max_length in
+      Thread_safe.block_on_async_exn (fun () ->
+          let%bind frontier = create_root_frontier ~max_length ~logger in
+          let genesis_root =
+            Transition_frontier.root frontier
+            |> Transition_frontier.Breadcrumb.transition_with_hash
+            |> With_hash.data
+            |> External_transition.forget_consensus_state_verification
+          in
+          let bootstrap =
+            Bootstrap_controller.For_tests.make_bootstrap ~logger
+              ~ancestor_prover ~genesis_root ~network ~max_length
+          in
+          let ledger_db =
+            Transition_frontier.For_tests.root_snarked_ledger frontier
+          in
+          let root_sync_ledger =
+            Root_sync_ledger.create ledger_db ~parent_log:logger
+          in
+          let parent_breadcrumb = Transition_frontier.best_tip frontier in
+          let breadcrumbs_gen =
+            Quickcheck_lib.gen_imperative_list
+              ( Quickcheck.Generator.return @@ Deferred.return
```

Comment:
This `Deferred.return` is unnecessary right? The `Deferred.all` immediately strips them, and this only produces filled deferreds.